### DISCUSSION TITLE
AllDay event startDate and endDate on android shown as the date before the appointment date

### DISCUSSION THREAD

**Discussion by sathe1295** [wmcmahan/react-native-calendar-events#391] [Q&A]
Hi,
I am trying to add an event in the calendar
start = "2021-10-30T05:00:00.000Z"
end = "2021-10-30T05:00:00.000Z" 

 details = {
            title: "All day event",
            startDate: start.ISOString(),
            endDate: end.toISOString(),
            notes: "Note for the event",
            description: "Note for the event",
            location: eventLocation,
            allDay: true,
            alarms: [{ date: Platform.OS === "ios" ? 60 : -60 }]
          }

Works fine on iOS. However on android, if allDay is true, the start and end date inside the event that gets added is shown as the date before for eg. w.r.t. above example, the date shown is Oct 29, whereas it should be Oct 30. 
I have changed the device timezone to match with the one in which I am creating the event. Can timezone be an issue?
Please help. Thanks in advance.

**Comment by LeandraH** [comment]
I don't know if this is still an issue for you but for allDay events, the start and end times need to be at midnight in Android. In this case, you would need to set them to be start = "2021-10-30T00:00:00.000Z" and end = "2021-10-31T00:00:00.000Z". Note that the end date is on midnight of the next day.
https://developer.android.com/reference/android/provider/CalendarContract.Events.html#writing-to-events
